### PR TITLE
Do not attempt to remove limits if keystoneAPI are not available

### PR DESCRIPTION
**Pull request by fmount**

### CODE REVIEW

File: controllers/glance_controller.go [hunk 1]
```diff
@@ -248,19 +248,20 @@ func (r *GlanceReconciler) reconcileDelete(ctx context.Context, instance *glance
...
-
-	if instance.IsQuotaEnabled() {
+	// If the CR is removed (Glance undeployed/cleaned up), we usually
+	// clean the resources created in the OpenStackControlPlane (e.g.,
+	// we remove the database from mariadb, delete the service and the
+	// endpoints in keystone). We should delete the limits created for
+	// the Glance service, and do not leave leftovers in the ctlplane.
+
+	// do not attempt to remove limits if keystoneAPI are not available
+	_, err = keystonev1.GetKeystoneAPI(ctx, helper, instance.Namespace, map[string]string{})
```
Comment (fmount) [inline controllers/glance_controller.go:258]: we already have finalizers related to both glance-{internal, external} endpoints and service, which means that we're not expecting keystone to be deleted before `Glance`.
Calling `GetKeystoneAPI` ensures the CR is still in the ctlplane.

**Comment by fmount** [comment]
Interesting that I don't see any issue with the quotas related test, but I see a new failure that I've never seen before [1]. 
Taking a note about it, and re-kicking functional tests for a yet another billion times (while I try to reproduce the new one
locally).

[1] https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openstack-k8s-operators_glance-operator/323/pull-ci-openstack-k8s-operators-glance-operator-main-functional/1707073962913042432/artifacts/test/build-log.txt

**Review by konan-abhi — APPROVED** [review]
Looks good to me, thank you!

**Pull request merged**